Foreman/DataCache.cs

```diff
@@ -1084,58 +1093,108 @@ private static Item FindOrCreateUnknownItem(String itemName)
+            LuaTable icons = ReadLuaLuaTable(values, "icons", true);
+            try
+            {
+                if (!Convert.ToString(values["name"]).Contains("barrel"))
+                {
+                    foreach (var iconKey in icons.Keys)
                     {
-                        //ErrorLogging.LogLine("recipe name: " + values["name"]);
-                        foreach (var s in icons.Values)
+                        if (iconKey != null)
                         {
-                            if (s is LuaTable)
+                            if (icons[iconKey] is LuaTable)
                             {
-                                //LuaTable tints = (LuaTable)((LuaTable)s)["tint"];
-                                //foreach (var k in tints.Keys)
-                                //{
-                                //    ErrorLogging.LogLine(Convert.ToString(k) + ": " + Convert.ToString(tints[k]));
-                                //}
-                                //ErrorLogging.LogLine(Convert.ToString(((LuaTable)s)["icon"]));
-                                fileName = Convert.ToString(((LuaTable)s)["icon"]);
+                                LuaTable iconTable = (LuaTable)icons[iconKey];
+                                int actualIconSize = Convert.ToInt32(Convert.ToDouble(iconTable["icon_size"]) * Convert.ToDouble(iconTable["scale"]));
+                                // create and scale image
+                                Bitmap iconImage = new Bitmap
+                                (
+                                    LoadImage(Convert.ToString(iconTable["icon"])),
+                                    actualIconSize,
+                                    actualIconSize
+                                );
+                                // tint image
+                                LuaTable tintTable = (LuaTable)iconTable["tint"];
+                                if (tintTable != null)
+                                {
+                                    Color tintColor = Color.FromArgb
+                                    (
+                                        Convert.ToInt32(Convert.ToDouble(tintTable["a"]) * 255),
+                                        Convert.ToInt32(Convert.ToDouble(tintTable["r"]) * 255),
+                                        Convert.ToInt32(Convert.ToDouble(tintTable["g"]) * 255),
+                                        Convert.ToInt32(Convert.ToDouble(tintTable["b"]) * 255)
+
+                                    );
+                                    for (int i = 0; i < actualIconSize; i++)
+                                    {
+                                        for (int j = 0; j < actualIconSize; j++)
+                                        {
+                                            if (iconImage.GetPixel(i, j).A != 0)
+                                            {
+                                                iconImage.SetPixel(i, j, tintColor);

```

Comment:
I'm not familiar with this API but is there not a "blend" function or something?

Reply:
not for `System.Drawing.Color` :(
but i'm not blending anyway, i just override the pixel if alpha is > 0, which shouldnt be to bad anyway
other option would be to convert the image to a `byteArray[][]` and do the operation there, but i just wanted to get it working first^^